### PR TITLE
Adding nightly NTO e2e test runs in ci-artifacts

### PR DESCRIPTION
**Pull request by jmencak**
This PR will enable running the NTO (upstream/release-4.8) e2e tests against 4.8 nightly builds of OCP using the `openshift-psap/ci-artifacts` repository.

### CODE REVIEW

File: ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nto.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+images:
+- dockerfile_path: build/Dockerfile
+  from: os
+  to: ci-artifacts
+releases:
+  latest:
+    candidate:
+      product: ocp
+      stream: nightly
+      version: "4.8"
```
Comment (kpouget) [inline ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nto.yaml:20]: I think you're running twice (master & release-4.8) on OCP 4.8
for GPU Operator I have no "nightly" on master, only presubmit

File: ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nto.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  steps:
+    cluster_profile: aws
+    test:
+    - as: nightly
+      cli: latest
+      commands: run nto e2e
+      credentials:
+      - mount_path: /var/run/psap-entitlement-secret
+        name: psap-entitlement-secret
+        namespace: test-credentials
```
Comment (kpouget) [inline ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nto.yaml:39]: credentials are not needed for NTO I think

Comment (jmencak) [reply ci-operator/config/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master__nto.yaml:39]: Agreed, will drop this. Thanks. :)

**Review by kpouget — CHANGES_REQUESTED** [review]
so,
one test `run nto e2e` on OCP 4.8 on presubmit (need to mark it as "do not always run")
one test `run nto e2e` on OCP 4.8 nightly at 23h
one generated test `nto-images`, I guess this one can also be marked as "do not always run"

and credentials are not necessary for NTO testing as you don't need entitlement or any secret

apart from these minor nits, looks good to me :)

... but I have to go, I'll let you ping someone else to merge it ;-)

File: ci-operator/jobs/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master-presubmits.yaml [hunk 3]
```diff
@@ -451,6 +451,133 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )images,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
[the remark this answers is not in the document]
Comment (jmencak) [reply ci-operator/jobs/openshift-psap/ci-artifacts/openshift-psap-ci-artifacts-master-presubmits.yaml]: Thanks for all the comments and help. Hopefully all addressed. Enjoy your time off!

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**